Mathlib/Algebra/Polynomial/Laurent.lean

```diff
@@ -512,6 +510,77 @@ theorem isLocalization : IsLocalization (Submonoid.powers (X : R[X])) R[T;T⁻¹
       rintro rfl
       exact ⟨1, rfl⟩ }
 
+theorem mk'_mul_T (p : R[X]) (n : ℕ) :
+  IsLocalization.mk' R[T;T⁻¹] p (⟨X^n,n,rfl⟩ : Submonoid.powers (X : R[X])) * T n =
+    toLaurent p := by
+  rw [←toLaurent_X_pow, ←algebraMap_eq_toLaurent, IsLocalization.mk'_spec, algebraMap_eq_toLaurent]
+
+theorem mk'_eq (p : R[X]) (n : ℕ) : IsLocalization.mk' R[T;T⁻¹] p
+  (⟨X^n,n,rfl⟩ : Submonoid.powers (X : R[X])) = toLaurent p * T (-n) := by
+  rw [←IsUnit.mul_left_inj (isUnit_T n), mul_T_assoc, neg_add_cancel, T_zero, mul_one]
+  exact mk'_mul_T p n
+
+theorem mk'_one_X_pow (n : ℕ) : IsLocalization.mk' R[T;T⁻¹] 1
+  (⟨X^n,n,rfl⟩ : Submonoid.powers (X : R[X])) = T (-n) := by
+  rw [mk'_eq 1 n, toLaurent_one, one_mul]
+
+theorem mk'_one_X : IsLocalization.mk' R[T;T⁻¹] 1
```

Comment:
These all seem like perfect simp lemmas.
Also, please add spaces after the commas inside anon constructors.

Otherwise, LGTM.


bors d+